packages/babel-generator/src/printer.js

```diff
@@ -538,8 +513,17 @@ export default class Printer {
       this._printedCommentStarts[comment.start] = true;
     }
 
-    // whitespace before
-    this.newline(this._whitespace ? this._whitespace.getNewlinesBefore(comment) : 0);
+    // We leave inline flow block comments inline
+    const isNonFlowBlockComment =
+      comment.type === "CommentBlock" &&
+      (
+        (!comment.value.startsWith(":") &&
+        !comment.value.startsWith("flow-include")) ||
```

Comment:
Do we have any tests for this? Does Flow actually require these to be on the same line, or is this just for making it nicer to read? I don't love having a special case like this.